planner/core/find_best_task.go

```diff
@@ -1265,13 +1265,14 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty, planCounter
 	return
 }
 
+// convertToIndexMergeScan builds the index merge scan for intersection or union cases.
 func (ds *DataSource) convertToIndexMergeScan(prop *property.PhysicalProperty, candidate *candidatePath, _ *physicalOptimizeOp) (task task, err error) {
 	if prop.IsFlashProp() || prop.TaskTp == property.CopSingleReadTaskType {
 		return invalidTask, nil
 	}
-	if prop.TaskTp == property.CopMultiReadTaskType && candidate.path.IndexMergeIsIntersection {
-		return invalidTask, nil
-	}
+	// lift the limitation of that double read can not build index merge **COP** task with intersection.
+	// that means we can output a cop task here without encapsulating it as root task, for the convenience of attaching limit to its table side.
+
 	if !prop.IsSortItemEmpty() && !candidate.isMatchProp {
 		return invalidTask, nil
 	}
```

Comment:
You need to return invalid if the prop has sort items for intersection cases.

Reply:
meaningful